core/src/main/java/org/apache/iceberg/avro/BuildAvroProjection.java

```diff
@@ -195,6 +196,8 @@ public Schema array(Schema array, Supplier<Schema> element) {
 
         // element was changed, create a new array
         if (elementSchema != array.getElementType()) {
+          // TODO: we do not copy field ids here. Probably not required,
+          //  but we do at other places in this class.
```

Comment:
This assumes that the incoming Avro schema has field IDs. The only reason to copy in other places is when we are building a replacement schema for some reason, like if the map didn't have the LogicalMap annotation (some old manifest files), or if the value schema is a subset because only some of its fields were projected.

Reply:
I've removed these todos.